### PR TITLE
fix: usage of `node12 which is deprecated` in CI

### LINKED ISSUE #1658: docs-check.yml `uses node12 which is deprecated`
https://github.com/huggingface/tokenizers/actions/runs/11322291257

### PR DESCRIPTION
**Pull request by hamirmahal**
fixes #1658.

### CODE REVIEW

File: .github/workflows/docs-check.yml [hunk 1]
```diff
@@ -14,18 +14,13 @@ jobs:
         uses: actions/checkout@v3
 
       - name: Install Python
-        uses: actions/setup-python@v1
+        uses: actions/setup-python@v5
```
Comment (hamirmahal) [inline .github/workflows/docs-check.yml:17]: `v5` is the latest major version of [`actions/setup-python`](https://github.com/actions/setup-python) at the time of this writing.

File: .github/workflows/docs-check.yml [hunk 2]
```diff
@@ -14,18 +14,13 @@ jobs:
...
         with:
           python-version: 3.9
 
       - name: Install dependencies
         run: pip install sphinx sphinx_rtd_theme setuptools-rust
 
-      - name: Install Rust
-        uses: actions-rs/toolchain@v1
-        with:
-          toolchain: stable
```
Comment (ArthurZucker) [inline .github/workflows/docs-check.yml:27]: is there a reason why this one is deleted as well?

**Review by ArthurZucker — COMMENTED** [review]
Thanks for updating!

File: .github/workflows/docs-check.yml (hunk 2 above)
Comment (hamirmahal) [reply .github/workflows/docs-check.yml:27]: Yes, that action also `uses node12 which is deprecated`.

**Comment by hamirmahal** [comment]
You're welcome!

**Comment by ArthurZucker** [comment]
the install dependencies for node build seems to break from this 😓

**Comment by hamirmahal** [comment]
It looks like https://github.com/hamirmahal/tokenizers/pull/1/commits/84e803122e2aebe477a63b9cdeac197c7cad4df3 fixes it.

At the very least, that change passes with all checks in a fork.

**Comment by ArthurZucker** [comment]
PR just merged to main fixed this! (it did all of the actions) 🤗 thanks for the contribution

**Comment by hamirmahal** [comment]
You're welcome! Feel free to close this if it's no longer necessary.

**Comment by ArthurZucker** [comment]
Closing then! 🤗

**Pull request closed without merging**